client/components/web-preview/content.jsx

```diff
@@ -0,0 +1,233 @@
+			debug( 'removing iframe contents' );
+			this.setIframeMarkup( '' );
+		}
+	}
+
+	setIframeMarkup( content ) {
+		if ( ! this.iframe ) {
+			debug( 'no iframe to update' );
+			return;
+		}
+		debug( 'adding markup to iframe', content.length );
+		this.iframe.contentDocument.open();
+		this.iframe.contentDocument.write( content );
+		this.iframe.contentDocument.close();
+	}
+
+	setIframeUrl( iframeUrl ) {
+		if ( ! this.iframe ) {
+			return;
+		}
+
+		// Bail if we've already set this url
+		if ( iframeUrl === this.state.iframeUrl ) {
+			return;
+		}
+
+		debug( 'setIframeUrl', iframeUrl );
+		try {
+			this.iframe.contentWindow.location.replace( iframeUrl );
+			this.setState( {
+				loaded: false,
+				iframeUrl: iframeUrl,
+			} );
+		} catch ( e ) {}
+	}
+
+	setDeviceViewport( device = 'computer' ) {
+		this.setState( { device } );
+
+		this.props.recordTracksEvent( 'calypso_web_preview_select_viewport_device', { device } );
+	}
+
+	setLoaded() {
+		if ( ! this.state.iframeUrl && ! this.props.previewMarkup ) {
+			debug( 'preview loaded, but nothing to show' );
+			return;
+		}
+		if ( this.props.previewMarkup ) {
+			debug( 'preview loaded with markup' );
+			this.props.onLoad( this.iframe.contentDocument );
+		} else {
+			debug( 'preview loaded for url:', this.state.iframeUrl );
+		}
+		this.setState( { loaded: true } );
+	}
+
+	render() {
+		const { translate } = this.props;
+
+		const className = classNames( this.props.className, 'web-preview__inner', {
+			'is-touch': this._hasTouch,
+			'is-with-sidebar': this.props.hasSidebar,
+			'is-visible': this.props.showPreview,
+			'is-computer': this.state.device === 'computer',
+			'is-tablet': this.state.device === 'tablet',
+			'is-phone': this.state.device === 'phone',
+			'is-seo': this.state.device === 'seo',
+			'is-loaded': this.state.loaded,
+		} );
+
+		return (
+			<div className={ className }>
+				<Toolbar setDeviceViewport={ this.setDeviceViewport }
+					device={ this.state.device }
+					{ ...this.props }
+					showExternal={ ( this.props.previewUrl ? this.props.showExternal : false ) }
+					showDeviceSwitcher={ this.props.showDeviceSwitcher && ! this._isMobile }
+					selectSeoPreview={ this.selectSEO }
+				/>
+				<div className="web-preview__placeholder">
+					{ ! this.state.loaded && 'seo' !== this.state.device &&
+						<div>
+							<Spinner />
+							{ this.props.loadingMessage &&
+								<span className="web-preview__loading-message">
+									{ this.props.loadingMessage }
+								</span>
+							}
+						</div>
+					}
+					<iframe
+						ref={ this.setIframeInstance }
+						className="web-preview__frame"
+						style={ { display: ( 'seo' === this.state.device ? 'none' : 'inherit' ) } }
+						src="about:blank"
+						onLoad={ this.setLoaded }
+						title={ this.props.iframeTitle || translate( 'Preview' ) }
+					/>
+					{ 'seo' === this.state.device &&
+						<SeoPreviewPane />
+					}
+				</div>
+			</div>
+		);
+	}
+}
+
+WebPreview.propTypes = {
```

Comment:
For now, I just copied `index.jsx` file and made few modifications here. I can definitely address this later, good idea